From 35b086a436d19837982232b5f91353c413a2ae5e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Mikko=20Salom=C3=A4ki?= <ms@datarespons.se>
Date: Thu, 6 Dec 2018 10:38:07 +0100
Subject: [PATCH] Remove all authenticated attributes from signature

---
 efi-updatevar.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/efi-updatevar.c b/efi-updatevar.c
index bc92cb8..bbea1e2 100644
--- a/efi-updatevar.c
+++ b/efi-updatevar.c
@@ -356,11 +356,11 @@ main(int argc, char *argv[])
 		BIO *bio = BIO_new_mem_buf(signbuf, signbuflen);
 		PKCS7 *p7 = PKCS7_sign(NULL, NULL, NULL, bio,
 				       PKCS7_BINARY | PKCS7_PARTIAL
-				       | PKCS7_DETACHED);
+				       | PKCS7_DETACHED | PKCS7_NOATTR);
 		const EVP_MD *md = EVP_get_digestbyname("SHA256");
 		PKCS7_sign_add_signer(p7, X, pkey, md, PKCS7_BINARY
-				      | PKCS7_DETACHED);
-		PKCS7_final(p7, bio, PKCS7_BINARY | PKCS7_DETACHED);
+				      | PKCS7_DETACHED | PKCS7_NOATTR);
+		PKCS7_final(p7, bio, PKCS7_BINARY | PKCS7_DETACHED | PKCS7_NOATTR);
 
 
 		int sigsize = i2d_PKCS7(p7, NULL);
